internal/server/proto/server.proto

```diff
@@ -2242,6 +2271,112 @@ message Release {
+* Status Reports
+********************************************************************/
+
+message UpsertStatusReportRequest {
+  // status report to upsert. If the id in the deployment is empty, then this
+  // will be an insert. Otherwise, this will be an update and if the ID
+  // isn't found, it will be an error.
+  StatusReport status_report = 1;
+}
+
+message UpsertStatusReportResponse {
+  // resulting object, you should replace this with what was sent in the request
+  // since the update operation may touch up the input data (i.e. update
+  // timestamps)
+  StatusReport status_report = 1;
+}
+
+message GetLatestStatusReportRequest {
+  // application that this belongs to
+  Ref.Application application = 1;
+
+  // workspace for the report, any workspace if empty
+  Ref.Workspace workspace = 2;
+}
+
+message ListStatusReportsRequest {
+  // The filters to apply to this request. These are ORed, so you should
+  // specify multiple filters in the StatusFilter for AND behavior.
+  repeated StatusFilter status = 1;
+
+  // Specifies the order of results. If this isn't specified, the results
+  // are in an undefined order.
+  OperationOrder order = 2;
+
+  // application that this release belongs to
+  Ref.Application application = 3;
+
+  // workspace that this should belong to. If this is empty, values in
+  // all workspaces will be listed.
+  Ref.Workspace workspace = 4;
+}
+
+
+message ListStatusReportsResponse {
+  // For 0.4.0, there should only the one 'latest' status report
+  repeated StatusReport status_reports = 1;
+}
+
+message GetStatusReportRequest {
+  Ref.Operation ref = 1;
+}
+
+// Not related to Status or StatusFilter messages
+//
+// StatusReport is the report genrated when querying the overall health of
+// a deployed or released application. This report can be either generated
+// by querying the platform itself which has performed the health checks,
+// or by Waypoint running user-defined health checks.
+message StatusReport {
+
+  // The application that this build is part of.
+  Ref.Application application = 1;
+
+  // The workspace that this exists in
+  Ref.Workspace workspace = 2;
+
+  // the reference id that is associated with this report
+  oneof target_id {
+    // the deployment id that this status report was generated on
+    string deployment_id = 3;
+
+    // the release id that this status report was generated on
+    string release_id = 4;
+  }
+
+  // status of the release operation.
+  // Note that this status is not related to the status report, but the current
+  // state of the StatusReport operation
+  Status status = 5;
+
+  // id is the unique ID for this status report.
+  string id = 6;
+
+  // StatusReport is the raw report object encoded directly from the plugin.
+  google.protobuf.Any status_report = 7;
+
+  // Resources that make up the report. The root most report is expected
+  // to be the overall health associated with the report.
```

Comment:
"root most report" is what? the zero index element? Perhaps clarify this comment a bit.